src/manifest/owned/pfm.rs

```diff
@@ -102,6 +81,49 @@ pub enum Element {
     },
 }
 
+/// asdf
+#[allow(missing_docs)]
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct AllowableFw {
+    #[cfg_attr(
+        feature = "serde",
+        serde(deserialize_with = "crate::serde::de_radix")
+    )]
+    pub version_count: u8,
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            deserialize_with = "crate::serde::de_bytestring",
+            serialize_with = "crate::serde::se_bytestring",
+        )
+    )]
+    pub firmware_id: Vec<u8>,
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            deserialize_with = "crate::serde::de_radix",
+            serialize_with = "crate::serde::se_bin",
+        )
+    )]
+    flags: u8,
+}
+
+impl AllowableFw {
+    /// Gets the `UpdatesTakeEffect` flag value.
+    pub fn get_updates_take_effect(&self) -> UpdatesTakeEffect {
+        UpdatesTakeEffect::from_u8(self.flags)
+    }
+
+    /// Sets the `UpdatesTakeEffect` flag value.
+    pub fn set_updates_take_effect(
+        &mut self,
+        updates_take_effect: UpdatesTakeEffect,
+    ) {
+        self.flags = updates_take_effect.set_bits_in_u8(self.flags);
```

Comment:
Maybe `set_bits` wants to take `&mut u8` instead?

Reply:
done